### PR TITLE
[Development] Gaze grasp action for Spot and Stretch

### PR DESCRIPTION
**Pull request by jimmytyyang**
## Motivation and Context

 Currently, Spot uses the model-based grasping in which if the camera can see the object, and the object is at the center of the frame, then the low-level API is called to grasp the object. The model-based grasping is also used in Stretch robot. 

## How Has This Been Tested

The following videos show how Stretch and Spot grasp the objects based on gaze action

https://user-images.githubusercontent.com/55121504/224075211-07827bad-6233-4111-8bea-a35a0dafcfa4.mp4

https://user-images.githubusercontent.com/55121504/224075216-404c26b4-468c-47d1-bd03-cf0b06ba283d.mp4

## Types of changes

- **\[Development\]** A pull request that add new features to the [habitat-lab](/habitat-lab) task and environment codebase. Development Pull Requests must be small (less that 500 lines of code change), have unit testing, very extensive documentation and examples. These are typically new tasks, environments, sensors, etc... The review process for these Pull Request is longer because these changes will be maintained by our core team of developers, so make sure your changes are easy to understand!

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes if required.

An illustration of the code
Screenshot 2023-03-09 at 10 34 49 AM

### CODE REVIEW

File: habitat-lab/habitat/config/benchmark/rearrange/play_stretch.yaml [hunk 1]
```diff
@@ -16,13 +16,16 @@ habitat:
...
+        grip_controller: "GazeGraspAction"
         arm_joint_mask: [1,0,0,0,1,1,1,1,1,1]
         arm_joint_dimensionality: 10
         grasp_thresh_dist: 0.15
         disable_grip: False
         delta_pos_limit: 0.0125
         ee_ctrl_lim: 0.015
+        gaze_distance_range: [0.1, 3.0]
+        center_cone_angle_threshold: 20.0
+        center_cone_vector: [0.0, 1.0, 0.0]
```
Comment (yvsriram) [inline habitat-lab/habitat/config/benchmark/rearrange/play_stretch.yaml:28]: Hi Jimmy! You may want to update the resolution and hfov of `head_panoptic_sensor` in this config.

File: habitat-lab/habitat/tasks/rearrange/actions/grip_actions.py [hunk 2]
```diff
@@ -168,3 +171,209 @@ def _grasp(self):
...
+        link_trans = self.cur_articulated_agent.sim_obj.get_link_scene_node(
+            cam_info.attached_link_id
+        ).transformation
+        # Get the camera offset transformation
+        offset_trans = mn.Matrix4.translation(cam_info.cam_offset_pos)
+        cam_trans = link_trans @ offset_trans @ cam_info.relative_transform
+
+        return cam_trans
+
+    def get_grasp_object_mask(self, abs_obj_idx):
```
Comment (yvsriram) [inline habitat-lab/habitat/tasks/rearrange/actions/grip_actions.py:235]: This function can be removed, right?

Comment (jimmytyyang) [reply habitat-lab/habitat/tasks/rearrange/actions/grip_actions.py]: Thank you @yvsriram! Good catch. Thank you so much for this!

File: habitat-lab/habitat/config/benchmark/rearrange/play_stretch.yaml (hunk 1 above)
Comment (jimmytyyang) [reply habitat-lab/habitat/config/benchmark/rearrange/play_stretch.yaml:28]: Thank you @yvsriram! Added `head_panoptic_sensor` to the config. Thank you so much for this!

**Review by yvsriram — APPROVED** [review]
LGTM, thank you!

**Pull request merged**